framework/gltf_loader.cpp

```diff
@@ -647,13 +648,19 @@ sg::Scene GLTFLoader::load_scene(int scene_index)
 				gltf_texture.name = images[gltf_texture.source]->get_name();
 			}
 
-			texture->set_sampler(*default_sampler);
+			// Get the properties for the image format. We'll need to check whether a linear sampler is valid.
+			const VkFormatProperties fmtProps = device.get_gpu().get_format_properties(images[gltf_texture.source]->get_format());
+
+			bool has_linear_filter = (fmtProps.optimalTilingFeatures & VK_FORMAT_FEATURE_SAMPLED_IMAGE_FILTER_LINEAR_BIT);
+
+			texture->set_sampler(has_linear_filter ? *default_sampler_linear : *default_sampler_nearest);
 		}
 
 		scene.add_component(std::move(texture));
 	}
 
-	scene.add_component(std::move(default_sampler));
+	scene.add_component(std::move(default_sampler_linear));
+	scene.add_component(std::move(default_sampler_nearest));
```

Comment:
Maybe only add the sampler(s), that are actually used.

Reply:
Done